src/Categories/Category/Monoidal/Reasoning.agda

```diff
@@ -137,3 +137,81 @@ split₂ˡ {f = f} {g} {h} = begin
   f ⊗₁ (g ∘ h)            ≈˘⟨ C.identityˡ ⟩⊗⟨refl ⟩
   (C.id ∘ f) ⊗₁ (g ∘ h)   ≈⟨ ⊗-distrib-over-∘ ⟩
   C.id ⊗₁ g ∘ f ⊗₁ h      ∎
+
+-- Combined splitting and re-association.
+
+module _ {X Y Z} {f : X ⇒ Z} {g : Y ⇒ Z} {h : X ⇒ Y} (f≈gh : f ≈ g ∘ h) where
+
+  infixr 4 split₁_⟩∘⟨_ split₂_⟩∘⟨_
+  infixl 5 _⟩∘⟨split₁_ _⟩∘⟨split₂_
+
+  split₁_⟩∘⟨_ : ∀ {V W} {i j : V ⇒ X ⊗₀ W} → i ≈ j →
+                f ⊗₁ C.id ∘ i ≈ g ⊗₁ C.id ∘ h ⊗₁ C.id ∘ j
+  split₁_⟩∘⟨_ {_} {_} {i} {j} i≈j = begin
+    f ⊗₁ C.id ∘ i                  ≈⟨ f≈gh ⟩⊗⟨refl ⟩∘⟨ i≈j ⟩
+    (g ∘ h) ⊗₁ C.id ∘ j            ≈⟨ split₁ˡ ⟩∘⟨refl ⟩
+    (g ⊗₁ C.id ∘ h ⊗₁ C.id) ∘ j    ≈⟨ C.assoc ⟩
+    g ⊗₁ C.id ∘ (h ⊗₁ C.id ∘ j)    ∎
+
+  split₂_⟩∘⟨_ : ∀ {V W} {i j : V ⇒ W ⊗₀ X} → i ≈ j →
+                C.id ⊗₁ f ∘ i ≈ C.id ⊗₁ g ∘ C.id ⊗₁ h ∘ j
+  split₂_⟩∘⟨_ {_} {_} {i} {j} i≈j = begin
+    C.id ⊗₁ f ∘ i                  ≈⟨ refl⟩⊗⟨ f≈gh ⟩∘⟨ i≈j ⟩
+    C.id ⊗₁ (g ∘ h) ∘ j            ≈⟨ split₂ˡ ⟩∘⟨refl ⟩
+    (C.id ⊗₁ g ∘ C.id ⊗₁ h) ∘ j    ≈⟨ C.assoc ⟩
+    C.id ⊗₁ g ∘ (C.id ⊗₁ h ∘ j)    ∎
+
+  _⟩∘⟨split₁_ : ∀ {V W} {i j : Z ⊗₀ W ⇒ V} → i ≈ j →
+                i ∘ f ⊗₁ C.id ≈ (j ∘ g ⊗₁ C.id) ∘ h ⊗₁ C.id 
+  _⟩∘⟨split₁_ {_} {_} {i} {j} i≈j = begin
+    i ∘ f ⊗₁ C.id                  ≈⟨ i≈j ⟩∘⟨ f≈gh ⟩⊗⟨refl ⟩
+    j ∘ (g ∘ h) ⊗₁ C.id            ≈⟨ refl⟩∘⟨ split₁ˡ ⟩
+    j ∘ (g ⊗₁ C.id ∘ h ⊗₁ C.id)    ≈⟨ C.sym-assoc ⟩
+    (j ∘ g ⊗₁ C.id) ∘ h ⊗₁ C.id    ∎
+
+  _⟩∘⟨split₂_ : ∀ {V W} {i j : W ⊗₀ Z ⇒ V} → i ≈ j →
+                i ∘ C.id ⊗₁ f ≈ (j ∘ C.id ⊗₁ g) ∘ C.id ⊗₁ h 
+  _⟩∘⟨split₂_ {_} {_} {i} {j} i≈j = begin
+    i ∘ C.id ⊗₁ f                  ≈⟨ i≈j ⟩∘⟨ refl⟩⊗⟨ f≈gh ⟩
+    j ∘ C.id ⊗₁ (g ∘ h)            ≈⟨ refl⟩∘⟨ split₂ˡ ⟩
+    j ∘ (C.id ⊗₁ g ∘ C.id ⊗₁ h)    ≈⟨ C.sym-assoc ⟩
+    (j ∘ C.id ⊗₁ g) ∘ C.id ⊗₁ h    ∎
+
+-- Combined merging and re-association.
+
+module _ {X Y Z} {f : Y ⇒ Z} {g : X ⇒ Y} {h : X ⇒ Z} (fg≈h : f ∘ g ≈ h) where
+
+  infixr 4 merge₁_⟩∘⟨_ merge₂_⟩∘⟨_
+  infixl 5 _⟩∘⟨merge₁_ _⟩∘⟨merge₂_
+
+  merge₁_⟩∘⟨_ : ∀ {V W} {i j : V ⇒ X ⊗₀ W} → i ≈ j →
+                f ⊗₁ C.id ∘ g ⊗₁ C.id ∘ i ≈ h ⊗₁ C.id ∘ j
+  merge₁_⟩∘⟨_ {_} {_} {i} {j} i≈j = begin
+    f ⊗₁ C.id ∘ (g ⊗₁ C.id ∘ i)    ≈⟨ C.sym-assoc ⟩
```

Comment:
It looks like all these proofs are the formal reverse of the ones for split - perhaps they could be simplified?

Reply:
Of course. Silly me. I'll fix these.